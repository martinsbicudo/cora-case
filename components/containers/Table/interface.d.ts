import { TransactionsType } from '@/context/TransactionsProvider/interface'

export interface TableProps {
  transactions: TransactionsType
}
