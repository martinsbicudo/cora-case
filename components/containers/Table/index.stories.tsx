import { StoryObj, Meta } from '@storybook/react'

import Table from '.'

export default {
  title: 'Containers / Table',
  component: Table,
} as Meta

export const Default: StoryObj = {}
