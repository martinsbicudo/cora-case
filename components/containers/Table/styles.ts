import styled from 'styled-components'

export const Table = styled.table`
  width: 100%;
  border-spacing: 0;
  padding-top: 30px;
  color: ${({ theme }) => theme.colors['gray-1']};
`

export const TableTr = styled.tr``

export const TableTrBottom = styled.tr``

export const TableTh = styled.th`
  text-align: left;
  font-weight: 400;
  font-size: 1.4rem;
  line-height: 2rem;

  &:first-child {
    visibility: hidden;
  }

  &:not(:first-child) {
    transform: translateY(-25px);
  }

  &:last-child {
    text-align: right;
    padding-right: 24px;
  }

  &:not(:last-child) {
    padding-left: 24px;
  }
`

export const TableTd = styled.td`
  text-align: left;
  font-weight: 400;
  font-size: 1.6rem;
  line-height: 2.4rem;
  padding-top: 20px;
  padding-bottom: 20px;
  border: 0 solid ${({ theme }) => theme.colors['gray-2']};

  &:first-child {
    color: ${({ theme }) => theme.colors['off-black']};
  }

  &:last-child {
    text-align: right;
    padding-right: 24px;
  }

  &:not(:last-child) {
    padding-left: 24px;
  }
`

export const TableTbody = styled.tbody`
  position: relative;
  border-radius: 16px;

  ${TableTr}:first-child,
  ${TableTr}:last-child {
    ${TableTd} {
      font-weight: 900;
      font-size: 1.2rem;
      line-height: 2rem;
      padding: 24px;
    }

    ${TableTd} span {
      font-weight: 400;
    }
  }

  ${TableTr}:first-child {
    ${TableTd} {
      position: absolute;
      top: 0;
      text-align: left;
      transform: translateY(-68px);
    }
  }

  ${TableTr}:not(:first-child):not(:last-child) {
    ${TableTd}:first-child {
      border-left-width: 1px;
    }

    ${TableTd}:last-child {
      border-right-width: 1px;
    }
  }

  ${TableTr}:nth-child(2) {
    ${TableTd} {
      border-top-width: 1px;
    }

    ${TableTd}:first-child {
      position: relative;
      border-top-left-radius: 16px;

      &:before {
        content: '';
        position: absolute;
        display: block;
        height: 16px;
        top: 0;
        left: 32px;
        width: 1px;
        background-color: #c7cbcf;
        transform: translateY(-100%);
      }
    }

    ${TableTd}:last-child {
      border-top-right-radius: 16px;
    }
  }

  ${TableTr}:nth-last-child(2) {
    ${TableTd} {
      border-bottom-width: 1px;
    }

    ${TableTd}:first-child {
      border-bottom-left-radius: 16px;
    }

    ${TableTd}:last-child {
      border-bottom-right-radius: 16px;
    }
  }

  &:not(:last-child) ${TableTr}:nth-last-child(2) {
    ${TableTd}:first-child {
      position: relative;

      &:after {
        content: '';
        position: absolute;
        display: block;
        height: 16px;
        bottom: 0;
        left: 32px;
        width: 1px;
        background-color: #c7cbcf;
        transform: translateY(100%);
      }
    }
  }
`
