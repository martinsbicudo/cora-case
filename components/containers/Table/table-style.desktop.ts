import { up } from 'styled-breakpoints'
import { css, StyledComponent, DefaultTheme } from 'styled-components'

const tableDesktop = (TableTd: StyledComponent<'td', DefaultTheme>) => css`
  ${up('md')} {
    position: relative;

    tr:first-child,
    tr:last-child {
      ${TableTd} {
        font-weight: 900;
        font-size: 1.2rem;
        line-height: 2rem;
        padding: 24px;
      }

      ${TableTd} span:first-child {
        font-weight: 400;
      }
    }

    tr:first-child {
      ${TableTd} {
        position: absolute;
        top: 0;
        text-align: left;
        transform: translateY(-68px);
      }
    }

    tr:not(:first-child):not(:last-child) {
      ${TableTd}:first-child {
        border-left-width: 1px;
      }

      ${TableTd}:last-child {
        border-right-width: 1px;
      }
    }

    tr:nth-child(2) {
      ${TableTd} {
        border-top-width: 1px;
      }

      ${TableTd}:first-child {
        position: relative;
        border-top-left-radius: 16px;

        &:before {
          content: '';
          position: absolute;
          display: block;
          height: 16px;
          top: 0;
          left: 32px;
          width: 1px;
          background-color: #c7cbcf;
          transform: translateY(-100%);
        }
      }

      ${TableTd}:last-child {
        border-top-right-radius: 16px;
      }
    }

    tr:nth-last-child(2) {
      ${TableTd} {
        border-bottom-width: 1px;
      }

      ${TableTd}:first-child {
        border-bottom-left-radius: 16px;
      }

      ${TableTd}:last-child {
        border-bottom-right-radius: 16px;
      }
    }

    &:not(:last-child) tr:nth-last-child(2) {
      ${TableTd}:first-child {
        position: relative;

        &:after {
          content: '';
          position: absolute;
          display: block;
          height: 16px;
          bottom: 0;
          left: 32px;
          width: 1px;
          background-color: #c7cbcf;
          transform: translateY(100%);
        }
      }
    }
  }
`

export default tableDesktop
