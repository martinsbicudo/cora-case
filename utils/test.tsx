import type { ReactElement, ReactNode } from 'react'

import { ThemeProvider } from 'styled-components'

import theme from '@/styles/theme'
import { render, RenderOptions } from '@testing-library/react'

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
export { customRender as render }
