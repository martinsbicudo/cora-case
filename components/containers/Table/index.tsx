import * as Styled from './styles'

const Header = () => {
  return (
    <Styled.Table>
      <thead>
        <Styled.TableTr>
          <Styled.TableTh>Nome</Styled.TableTh>
          <Styled.TableTh>Tipo de transação</Styled.TableTh>
          <Styled.TableTh>Data</Styled.TableTh>
          <Styled.TableTh>Valor</Styled.TableTh>
        </Styled.TableTr>
      </thead>
      <Styled.TableTbody>
        <Styled.TableTr>
          <Styled.TableTd colSpan={4}>23 de Agosto</Styled.TableTd>
        </Styled.TableTr>
        <Styled.TableTr>
          <Styled.TableTd>Test</Styled.TableTd>
          <Styled.TableTd>Tansferência recebida</Styled.TableTd>
          <Styled.TableTd>Hoje - 12 Jun 2020 - 17:35</Styled.TableTd>
          <Styled.TableTd>+ R$ 320,00</Styled.TableTd>
        </Styled.TableTr>
        <Styled.TableTr>
          <Styled.TableTd colSpan={4}>
            <span>saldo</span> do dia R$ 3.110,08
          </Styled.TableTd>
        </Styled.TableTr>
      </Styled.TableTbody>
      <Styled.TableTbody>
        <Styled.TableTr>
          <Styled.TableTd colSpan={4}>23 de Agosto</Styled.TableTd>
        </Styled.TableTr>
        <Styled.TableTr>
          <Styled.TableTd>Test</Styled.TableTd>
          <Styled.TableTd>Tansferência recebida</Styled.TableTd>
          <Styled.TableTd>Hoje - 12 Jun 2020 - 17:35</Styled.TableTd>
          <Styled.TableTd>+ R$ 320,00</Styled.TableTd>
        </Styled.TableTr>
        <Styled.TableTr>
          <Styled.TableTd colSpan={4}>
            <span>saldo</span> do dia R$ 3.110,08
          </Styled.TableTd>
        </Styled.TableTr>
      </Styled.TableTbody>
    </Styled.Table>
  )
}

export default Header
