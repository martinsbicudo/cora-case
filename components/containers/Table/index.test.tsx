import { render } from '@/test/helpers'

import Table from '.'

describe('Containers / Table', () => {
  const renderComponent = () => render(<Table />)

  it('should render component', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
