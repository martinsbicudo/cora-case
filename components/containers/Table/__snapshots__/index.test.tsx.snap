// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers / Table should render component 1`] = `
<div>
  <table
    class="sc-bcXHqe hqVljI"
  >
    <thead>
      <tr
        class="sc-gswNZR"
      >
        <th
          class="sc-hLBbgP iFPXzE"
        >
          Nome
        </th>
        <th
          class="sc-hLBbgP iFPXzE"
        >
          Tipo de transação
        </th>
        <th
          class="sc-hLBbgP iFPXzE"
        >
          Data
        </th>
        <th
          class="sc-hLBbgP iFPXzE"
        >
          Valor
        </th>
      </tr>
    </thead>
    <tbody
      class="sc-jSUZER gaMrdr"
    >
      <tr
        class="sc-gswNZR"
      >
        <td
          class="sc-eDvSVe dbMYCm"
          colspan="4"
        >
          23 de Agosto
        </td>
      </tr>
      <tr
        class="sc-gswNZR"
      >
        <td
          class="sc-eDvSVe dbMYCm"
        >
          Test
        </td>
        <td
          class="sc-eDvSVe dbMYCm"
        >
          Tansferência recebida
        </td>
        <td
          class="sc-eDvSVe dbMYCm"
        >
          Hoje - 12 Jun 2020 - 17:35
        </td>
        <td
          class="sc-eDvSVe dbMYCm"
        >
          + R$ 320,00
        </td>
      </tr>
      <tr
        class="sc-gswNZR"
      >
        <td
          class="sc-eDvSVe dbMYCm"
          colspan="4"
        >
          <span>
            saldo
          </span>
           do dia R$ 3.110,08
        </td>
      </tr>
    </tbody>
    <tbody
      class="sc-jSUZER gaMrdr"
    >
      <tr
        class="sc-gswNZR"
      >
        <td
          class="sc-eDvSVe dbMYCm"
          colspan="4"
        >
          23 de Agosto
        </td>
      </tr>
      <tr
        class="sc-gswNZR"
      >
        <td
          class="sc-eDvSVe dbMYCm"
        >
          Test
        </td>
        <td
          class="sc-eDvSVe dbMYCm"
        >
          Tansferência recebida
        </td>
        <td
          class="sc-eDvSVe dbMYCm"
        >
          Hoje - 12 Jun 2020 - 17:35
        </td>
        <td
          class="sc-eDvSVe dbMYCm"
        >
          + R$ 320,00
        </td>
      </tr>
      <tr
        class="sc-gswNZR"
      >
        <td
          class="sc-eDvSVe dbMYCm"
          colspan="4"
        >
          <span>
            saldo
          </span>
           do dia R$ 3.110,08
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
